internal/services/logz/logz_sub_account_resource.go

```diff
@@ -0,0 +1,244 @@
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation of the %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+	return resourceLogzSubAccountRead(d, meta)
+}
+
+func resourceLogzSubAccountRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Logz.SubAccountClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogzSubAccountID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.MonitorName, id.AccountName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] logz %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	d.Set("name", id.AccountName)
+	d.Set("logz_monitor_id", parse.NewLogzMonitorID(id.SubscriptionId, id.ResourceGroup, id.MonitorName).ID())
+
+	if props := resp.Properties; props != nil {
+		d.Set("enabled", props.MonitoringStatus == logz.MonitoringStatusEnabled)
+		d.Set("user", d.Get("user"))
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceLogzSubAccountUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Logz.SubAccountClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogzSubAccountID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	props := logz.MonitorResourceUpdateParameters{
+		Properties: &logz.MonitorUpdateProperties{},
+	}
+
+	if d.HasChange("enabled") {
+		monitoringStatus := logz.MonitoringStatusDisabled
+		if d.Get("enabled").(bool) {
+			monitoringStatus = logz.MonitoringStatusEnabled
+		}
+		props.Properties.MonitoringStatus = monitoringStatus
+	}
+
+	if d.HasChange("tags") {
+		props.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
+	}
+
+	if _, err := client.Update(ctx, id.ResourceGroup, id.MonitorName, id.AccountName, &props); err != nil {
+		return fmt.Errorf("updating %s: %+v", id, err)
+	}
+
+	return resourceLogzSubAccountRead(d, meta)
+}
+
+func resourceLogzSubAccountDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Logz.SubAccountClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogzSubAccountID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, id.ResourceGroup, id.MonitorName, id.AccountName)
+	if err != nil {
+		return fmt.Errorf("deleting %s: %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for deletion of the %s: %+v", id, err)
+	}
+
+	// API has bug, which appears to be eventually consistent. Tracked by this issue: https://github.com/Azure/azure-rest-api-specs/issues/18572
```

Comment:
This resource has the same issue with [this existing resource](https://github.com/hashicorp/terraform-provider-azurerm/blob/cfd2651748d16f2a825b588d46fd3050a4ea65dd/internal/services/compute/dedicated_host_resource.go#L255-L292). I use the same way to address this issue. The reason is: Since the same Azure Resource is deployed in multiple regions, after WaitForCompletionRef returns, it only means the resource data of one region is deleted, however, the data cache may still exist in other regions. We have to run multiple GET operations to test whether data in other regions still exist. Only after the data cache in all regions are deleted, we can then delete its parent resource.